src/main/java/soot/jimple/toolkits/callgraph/ClinitElimAnalysis.java

```diff
@@ -79,13 +81,22 @@ protected FlowSet<SootMethod> entryInitialFlow() {
 
   @Override
   protected FlowSet<SootMethod> newInitialFlow() {
-    HashSparseSet<SootMethod> set = new HashSparseSet<SootMethod>();
+    HashSparseSet<SootMethod> returnedFlowSet = new HashSparseSet<>();
```

Comment:
will this method be called multiple times? I thought it is only called once at the beginning.

Reply:
It seems it is called multiple times for the same node. 
We see this by checking the CPU time spent in this method. For the test subject, we see that Soot spends almost 9 % of the CPU time in this method. 
After adding this cache mechanism, the CPU time of this method has almost no impact ... less than 1 % 

<img width="1278" alt="afterPR" src="https://github.com/soot-oss/soot/assets/10850220/4973f490-57be-47c6-9a92-e3572c784c73">
<img width="957" alt="beforePR" src="https://github.com/soot-oss/soot/assets/10850220/049d83f6-9ca7-4063-9d32-753f91315af5">

Reply:
It seems it is called multiple times for the same node.
We see this by checking the CPU time spent in this method. For the test subject, we see that Soot spends almost 9 % of the CPU time in this method.
After adding this cache mechanism, the CPU time of this method has almost no impact ... less than 1 %

Reply:
<img width="718" alt="afterPR" src="https://github.com/soot-oss/soot/assets/10850220/79bea38f-2f73-49f9-a2c1-2774a2d0be6d">
<img width="908" alt="beforePR" src="https://github.com/soot-oss/soot/assets/10850220/f11c63b2-0938-4e64-90cc-5efe475f79bc">